### PR TITLE
[website_customer] Migrated to new api.

### PR DESCRIPTION
**Pull request by rgo-odoo**
Task: https://www.odoo.com/web#id=12026&view_type=form&model=project.task&action=327
Pad: https://pad.odoo.com/p/r.TkQhEhvIYXjGJx6w

### CODE REVIEW

File: addons/website_customer/__openerp__.py [hunk 1]
```diff
@@ -17,13 +17,13 @@
...
+        'data/res_partner_demo.xml',
     ],
     'data': [
-        'views/website_customer.xml',
-        'views/website_customer_view.xml',
+        'views/website_customer_templates.xml',
+        'views/res_partner_views.xml',
         'security/ir.model.access.csv',
-        'security/ir_rule.xml',
+        'security/res_partner_tag_security.xml',
```
Comment (msh-odoo) [inline addons/website_customer/__openerp__.py]: Arrange this data list in proper order

File: addons/website_customer/controllers/main.py [hunk 2]
```diff
@@ -45,23 +44,19 @@ def customers(self, country_id=0, page=0, country_name='', tag_id=0, **post):
...
-            cr, openerp.SUPERUSER_ID, domain, ["id", "country_id"],
-            groupby="country_id", orderby="country_id", context=request.context)
-        country_count = partner_obj.search(
-            cr, openerp.SUPERUSER_ID, domain, count=True, context=request.context)
+        countries = PartnerSudo.read_group(domain, ["id", "country_id"], groupby="country_id", orderby="country_id")
+        country_count = sum(country_dict['country_id_count'] for country_dict in countries)
 
         if country_id:
             domain += [('country_id', '=', country_id)]
+            curr_country = request.env['res.country'].browse(country_id).exists()
```
Comment (msh-odoo) [inline addons/website_customer/controllers/main.py]: Is there a reason to use exists ?

File: addons/website_customer/controllers/main.py [hunk 3]
```diff
@@ -69,7 +63,7 @@ def customers(self, country_id=0, page=0, country_name='', tag_id=0, **post):
         })
 
         # search customers to display
-        partner_count = partner_obj.search_count(cr, openerp.SUPERUSER_ID, domain, context=request.context)
+        partner_count = PartnerSudo.search_count(domain)
```
Comment (msh-odoo) [inline addons/website_customer/controllers/main.py]: We can put search here, as we are already using search at line 77 and then len(partners)

File: addons/website_customer/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class ResPartnerTags(models.Model):
+    _description = 'Partner Tags - These tags can be used on website to find customers by sector, or ... '
+    _name = 'res.partner.tag'
+    _inherit = 'website.published.mixin'
+
+    name = fields.Char(string='Category Name', required=True, translate=True)
+    partner_ids = fields.Many2many('res.partner', 'res_partner_res_partner_tag_rel', 'tag_id', 'partner_id', string='Partners')
+    classname = fields.Selection('get_selection_class', string='Class', help="Bootstrap class to customize the color of the tag", required=True, default='default')
```
Comment (msh-odoo) [inline addons/website_customer/models/res_partner.py]: I would keep it something like:
``` classname = fields.Selection(get_selection_class, string='Class', help="Bootstrap class to customize the color of the tag", required=True, default='default') ```

and put get_selection_class above field definition

**Comment by tde-banana-odoo** [comment]
Done in master at https://github.com/odoo/odoo/commit/3275a90ff0ee71292b32a6005d2b1a289a09154b . Thanks !

**Pull request closed without merging**